packages/framework/src/workflow.ts

```diff
@@ -27,9 +31,51 @@ export function workflow<
   workflowId: string,
   execute: Execute<T_Payload, T_Control>,
   workflowOptions?: WorkflowOptions<T_PayloadSchema, T_ControlSchema>
-): DiscoverWorkflowOutput {
+): Workflow<T_Payload> {
   const options = workflowOptions ? workflowOptions : {};
 
+  const apiClient = initApiClient(process.env.NOVU_SECRET_KEY as string);
+
+  const trigger: Workflow<T_Payload>['trigger'] = async (event) => {
+    if (!process.env.NOVU_SECRET_KEY) {
+      throw new MissingSecretKeyError();
+    }
+
+    let validatedData: T_Payload;
+    if (options.payloadSchema) {
+      const validationResult = await validateData(options.payloadSchema, event.payload);
+      if (validationResult.success === false) {
+        throw new WorkflowPayloadInvalidError(workflowId, validationResult.errors);
+      }
+      validatedData = validationResult.data;
+    }
+    const bridgeUrl = await getBridgeUrl();
+
+    const requestPayload = {
+      name: workflowId,
+      to: event.to,
+      payload: {
+        ...event?.payload,
+      },
+      ...(event.transactionId && { transactionId: event.transactionId }),
+      ...(event.overrides && { overrides: event.overrides }),
+      ...(event.actor && { actor: event.actor }),
+      ...(event.tenant && { tenant: event.tenant }),
+      ...(bridgeUrl && { bridgeUrl }),
+    };
+
+    const result = await apiClient.post<EventTriggerResponse>('/events/trigger', requestPayload);
+
+    const cancel = async () => {
+      return apiClient.delete<CancelEventTriggerResponse>(`/events/trigger/${result.transactionId}`);
+    };
```

Comment:
Adding [cancel](https://docs.novu.co/api-reference/events/cancel-triggered-event) for a complete E2E story on the `trigger`.